### PR TITLE
`dev-cmd/audit`: fix cask arg handling

### PR DESCRIPTION
**Pull request by Rylan12**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This fixes an issue where `brew audit --cask --new-cask` did not run the [`check_token_conflicts`](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/cask/audit.rb#L479) check as it should have. This was because at some point in the past, `args.token_conflicts?` started returning `true`/`false` instead of `true`/`nil`. Since `false.nil?` is `false` but `false.blank?` is `true`, this wasn't triggering properly. There are a few other implied audit rules that likely were affected as a result of this. that I've fixed here, too.

### CODE REVIEW

File: Library/Homebrew/cask/audit.rb [hunk 1]
```diff
@@ -27,15 +27,16 @@ def initialize(cask, appcast: nil, download: nil, quarantine: nil,
...
+      online = new_cask if online.blank?
+      strict = new_cask if strict.blank?
 
       # `online` implies `appcast` and `download`
-      appcast = online if appcast.nil?
-      download = online if download.nil?
+      appcast = online if appcast.nil? # `appcast` is `nil` if neither `--appcast` nor `--no-appcast` are passed
+      download = online if download.blank?
 
+      odie online
```
Comment (carlocab) [inline Library/Homebrew/cask/audit.rb:37]: 🙃

Comment (Rylan12) [reply Library/Homebrew/cask/audit.rb:37]: Lol oops let's not commit that 😅

```suggestion
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by Rylan12** [comment]
I realized that it _is_ valuable to have a distinction between `nil` and `false` for testing. That way, `false` means "this is 100% false never set it to `true`" and `nil` means "unset so make whatever changes are needed."

Now, we can set `new_cask` to `true` while keeping `online` off. Before, setting `online` to `false` was treated as if `--online` wasn't passed (because there's no `--no-online` flag) and so the `--new` was overriding the value set in the test.

Instead, we handle the change in the `brew audit` command, itself, by adding `|| nil` to pass `nil` (i.e. "unset") to the auditor whenever `args.foo?` returns `false`.

In other words, the argument parser returns `false` when a (non-boolean) switch isn't set, and the auditor expects `nil` to mean unset. So, we change `false`s to `nil`s before passing them.

---

`--[no-]appcast` is excluded from all of the above because the argument parser already returns `nil`/`false`/`true` because it's a boolean switch.
